src/supervisor/metadata-extractor.ts

```diff
@@ -215,9 +217,20 @@ export async function buildMetadataForWorkload(
   if (podOwner === undefined) {
```

Comment:
I wonder if to make this change a bit safer, we could also check if the pod.ownerReferences contain kind: Node? This way we fix this issue only for the specific static Pods case, without risking potential other bugs when the Pod has a regular workload as a parent but we couldn't retrieve it